sbteclipse-core/src/main/scala/com/typesafe/sbteclipse/core/Eclipse.scala

```diff
@@ -441,19 +442,26 @@ private object Eclipse extends EclipseSDTConfig {
     ) reduceLeft (_ +++ _)
   }
 
-  def scalacOptions(ref: ProjectRef, state: State): Validation[Seq[(String, String)]] =
+  def scalacOptions(ref: ProjectRef, state: State): Validation[Seq[(String, String)]] = {
     // Here we have to look at scalacOptions *for compilation*, vs. the ones used for testing.
     // We have to pick one set, and this should be the most complete set.
-    evaluateTask(Keys.scalacOptions in sbt.Compile, ref, state) map (options =>
-      if (options.isEmpty)
-        Nil
-      else {
-        fromScalacToSDT(options) match {
-          case Seq() => Seq()
-          case options => ("scala.compiler.useProjectSettings" -> "true") +: options
-        }
-      }
-    )
+    (evaluateTask(Keys.scalacOptions in sbt.Compile, ref, state) |@| setting(Keys.scalaVersion in sbt.Compile, state)) { (options, version) =>
```

Comment:
I need to scope the `scalaVersion` to the project's `ref`, otherwise it's using the Global value (which is currently 2.10.4).